lib/table_sync/publisher.rb

```diff
@@ -4,26 +4,27 @@ class TableSync::Publisher < TableSync::BasePublisher
   DEBOUNCE_TIME = 1.minute
 
   # 'original_attributes' are not published, they are used to resolve the routing key
-  def initialize(object_class, original_attributes, destroyed: nil, confirm: true, state: :updated)
+  def initialize(object_class, original_attributes, **opts)
     @object_class = object_class.constantize
     @original_attributes = filter_safe_for_serialization(original_attributes.deep_symbolize_keys)
-    @confirm = confirm
+    @confirm = opts[:confirm].nil? ? true : opts[:confirm]
+    @debounce_time = opts[:debounce_time].nil? ? nil : opts[:debounce_time].seconds
 
-    if destroyed.nil?
-      @state = validate_state(state)
+    if opts[:destroyed].nil?
+      @state = validate_state(opts[:state])
     else
       # TODO Legacy job support, remove
-      @state = destroyed ? :destroyed : :updated
+      @state = opts[:destroyed] ? :destroyed : :updated
     end
   end
 
   def publish
-    return enqueue_job if destroyed?
+    return enqueue_job if destroyed? || debounce_time_zero?
 
-    sync_time = Rails.cache.read(cache_key) || current_time - DEBOUNCE_TIME - 1.second
+    sync_time = Rails.cache.read(cache_key) || current_time - debounce_time_offset - 1.second
     return if sync_time > current_time
 
-    next_sync_time = sync_time + DEBOUNCE_TIME
+    next_sync_time = sync_time + debounce_time_offset
```

Comment:
Maybe add 1 second that were subtracted in `sync_time `?

Reply:
Should we add extra second? I've check original code and didn't found any extra calculation there.